cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -0,0 +1,1319 @@
+        description:
+            - When removing a VM using state 'absent', also remove any network interfaces associate with the VM.
+        default: true
+        aliases:
+            - delete_nics
+        required: false
+    delete_virtual_storage:
+        description:
+            - When removing a VM using state 'absent', also remove any storage blobs associated with the VM.
+        default: true
+        aliases:
+            - delete_vhd
+        required: false
+    delete_public_ips:
+        description:
+            - When removing a VM using state 'absent', also remove any public IP addresses associate with the VM.
+        default: true
+        required: false
+    tags:
+        description:
+            - "Dictionary of string:string pairs to assign as metadata to the object. Metadata tags on the object
+              will be updated with any provided values. To remove tags use the purge_tags option."
+        default: null
+        required: false
+    purge_tags:
+        description:
+            - Use to remove tags from an object. Any tags not found in the tags parameter will be removed from
+              the object's metadata.
+        default: false
+        required: false
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Chris Houseknecht (@chouseknecht)"
+    - "Matt Davis (@nitzmahone)"
+
+'''
+EXAMPLES = '''
+
+- name: Create VM with defaults
+  azure_rm_virtualmachine:
+    resource_group: Testing
+    name: testvm10
+    admin_username: chouseknecht
+    admin_password: <your password here>
+    image:
+      offer: CentOS
+      publisher: OpenLogic
+      sku: '7.1'
+      version: latest
+
+- name: Create a VM with exiting storage account and NIC
+  azure_rm_virtualmachine:
+    resource_group: Testing
+    name: testvm002
+    vm_size: Standard_D4
+    storage_account: testaccount001
+    admin_username: adminUser
+    ssh_public_keys:
+      path: /home/adminUser/.ssh/authorized_keys
+      key_data: < insert yor ssh public key here... >
+    network_interfaces: testvm001
+    image:
+      offer: CentOS
+      publisher: OpenLogic
+      sku: '7.1'
+      version: latest
+
+- name: Power Off
+  azure_rm_virtualmachine:
+    resource_group: Testing
+    name: testvm002
+    state: stopped
+
+- name: Deallocate
+  azure_rm_virtualmachine:
+    resource_group: Testing
+    name: testvm002
+    state: stopped
+    deallocate: yes
+
+- name: Power On
+  azure_rm_virtualmachine:
+    resource_group:
+    name: testvm002
+
+- name: Restart
+  azure_rm_virtualmachine:
+    resource_group:
+    name: testvm002
+    restart: yes
+
+'''
+
+RETURN = '''
+changed:
+    description: Whether or not the object was changed.
+    returned: always
+    type: bool
+    sample: True
+actions:
+    description: List of descriptive actions performed by the module.
+    returned: always
+    type: list
+    sample: [
+        "Powered on virtual machine testvm10"
+    ]
+differences:
```

Comment:
return 'diff' key instead of 'differences' and also make it conditional to the module's '_diff' property